src/BaGet.Core/Services/FileStorageService.cs

```diff
@@ -96,13 +99,19 @@ public Task DeleteAsync(string path, CancellationToken cancellationToken = defau
 
         private string GetFullPath(string path)
         {
-            // TODO: This should check that the result is in _storePath for security.
             if (string.IsNullOrEmpty(path))
             {
                 throw new ArgumentException("Path is required", nameof(path));
             }
 
-            return Path.Combine(_storePath, path);
+            // Verify path is under the _storePath.
+            string fullPath = Path.GetFullPath(Path.Combine(_storePath, path));
+            if (!fullPath.StartsWith(_storePath))
+            {
+                throw new ArgumentException("Path resolves outside store path");
```

Comment:
Could you add a `nameof(path)` to be consistent with line 104?

Reply:
Also please add a unit test in [`FileStorageServiceTests.cs`](https://github.com/loic-sharma/BaGet/blob/master/tests/BaGet.Core.Tests/Services/FileStorageServiceTests.cs)